src/TraceEvent/Computers/TraceProcess.cs

```diff
@@ -138,9 +138,17 @@ public static void SetupCallbacks(TraceEventDispatcher source)
         }
 
         private static TraceEventDispatcher m_currentSource; // used to verify non-concurrent usage
+        
+        internal static void AddCallbackOnProcessCleanup(this TraceEventDispatcher source, Action<TraceProcess> OnProcessRemoved)
```

Comment:
We already have a public AddCallbackOnProcessStart, and AddCallbackOnProcessCleanup is really just the flip side.  We should rename this method to AddCallbackOnProcessStart can make it public (and move it next to the other method).

Reply:
Done.  The one thing I am considering now is that the semantic of Stop to me says that the callback should be fired once we receive the EndGroup event.  Right now the code is structured to only fire this event if it is purged due to lifetime - that feels like it breaks the Stop contract.  I will think a bit more about this.

Reply:
This is an good point, and brings up other issues.    Processes can be very long lived, and I don't think we should be purging unterminated processes by default.     I think by default we should only remove things we DO have stops for.  

Now this means that we can still have leaks, but only if our process monitoring is got bugs where we 'miss' the termination of processes.    It is not clear this is important, and if it is I think we should have some sort of more explicit mechanism.

Reply:
I changed the semantic to fire the event at Stop and only purge if a process was stopped.  I think this is fine, because this Lifetime is paired with the Dispatch hook only processes that matter will be analyzed.  If Dispatch hook is enabled then all live processes will be retained.  The big scenario here was for xmlconfiguration.exe which runs and stops all the time.  The current code addresses that case.

Reply:
I like that.